src/app/user/add.tsx

```diff
@@ -0,0 +1,90 @@
+import { View } from 'react-native';
+import { ControlledInput } from '_components/Input/ControlledInput';
+import { z } from 'zod';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Label } from '_components/Label/StyledLabel';
+import { Button } from '_components/Button/Button';
+import { createMutation } from 'react-query-kit';
+import { UserData } from '_types/Users';
+import { reqresApi } from '_utils/api';
+import { useQueryClient } from '@tanstack/react-query';
+import { useUsers } from '_hooks/useUsers';
+import { useNavigation } from 'expo-router';
+
+const schema = z.object({
+  firstName: z.string().min(2),
+  lastName: z.string().min(2),
+  email: z.string().email(),
+});
+
+const DEFAULT_VALUES = {
+  firstName: '',
+  lastName: '',
+  email: '',
+};
+
+const useUserMutation = createMutation<
```

Comment:
Shouldn't we extract useUserMutation here ? Im not sure bcs of the concern of separation do you think we would use this somewhere else ?